### DISCUSSION TITLE
Ability to disable the 'not-prose' functionality in the typography plugin

### DISCUSSION THREAD

**Discussion by seeARMS** [tailwindlabs/tailwindcss#7684] [Ideas]
I'm using @tailwindcss/typography along with Maizzle, to craft emails. I'm using sendgrid to send these emails. I'm using the `prose` functionality in the `@tailwindcss/typography` plugin, in order to ensure the emails look great.

There are a handful of CSS elements that Maizzle produces, when used with `@tailwindcss/typography`, which strip all CSS from the resulting email. For example, if `ol[type="a" s]` is included in the email's CSS, the email style gets stripped.

I've been getting around this by modifying the Tailwind typography plugin defaults to disable these styles:

```
module.exports = {
  [...]
    extend: {
      typography: (theme) => ({
        DEFAULT: {
          css: {
            color: theme("colors.gray.800"),
            'ol[type="A"]': false,
            'ol[type="a"]': false,
            'ol[type="A" s]': false,
            'ol[type="a" s]': false,
            'ol[type="I"]': false,
            'ol[type="i"]': false,
            'ol[type="I" s]': false,
            'ol[type="i" s]': false,
            'ol[type="1"]': false,
          },
        },
      }),
}
```
This worked great.

I upgraded to the latest Typography version (0.5.2), and there's a new addition of the `not-prose` class to disable prose styles selectively. The resulting CSS in the email that Maizzle produces now has these classes:

```
      ul ol,
      ol ul,
      ol ol):not(:where([class~="not-prose"] *)) {
        margin-top: 0.75em;
        margin-bottom: 0.75em;
      }

      ul ol,
      ol ul,
      ol ol):not(:where([class~="not-prose"] *)) {
        margin-top: 0.8888889em;
        margin-bottom: 0.8888889em;
      }
```

This causes the email CSS to break:

<img width="460" alt="image" src="https://user-images.githubusercontent.com/3957773/155856469-eba7828c-a54f-4f57-ab9b-a8be44fd0fc9.png">

Removing this CSS from the email, and re-sending it, causes the styles to look completely normal:

<img width="637" alt="image" src="https://user-images.githubusercontent.com/3957773/155856555-4dbbf153-63fd-47dd-a55c-e400cef9d8cf.png">

It doesn't appear possible to disable the `not-prose` classes from being generated. There's nothing in the [defaults](https://github.com/tailwindlabs/tailwindcss-typography/blob/master/src/styles.js) to override. It'd be great to add functionality to disable this.

**Comment by adamwathan** [comment]
Hey! Try the `target: “legacy”` option:

https://github.com/tailwindlabs/tailwindcss-typography/blob/master/src/index.test.js#L335

Sort of a secret API, have kept it quiet just to get a better sense of how many people actually need it by waiting for issues like this to pop up.

**Reply by seeARMS** [reply]
Ahh - that's perfect. Thank you!

**Comment by changethe** [comment]
thanks, this was really helpful!

I was trying the whole day to get purgecss to remove all the bloat from the typography plugin, but it seems to have trouble with the conditional syntax.

activating the legacy option solved it.

**Comment by archonic** [comment]
I believe the legacy option will fix issues I'm having trying to implement `.prose` with Tiptap but I'm not sure how to activate it with vite.
